tools/python/boutiques/localExec.py

```diff
@@ -132,8 +132,21 @@ def __init__(self, desc, invocation, options={}):
         self.desc_path = desc    # Save descriptor path
         self.errs = []        # Empty errors holder
         self.invocation = invocation
-        # Parse JSON descriptor
-        self.desc_dict = loadJson(desc)
+
+        # Extra Options
+        # Include: forcePathType and debug
+        self.debug = False
+        self.zenodo = False
+        for option in list(options.keys()):
+            setattr(self, option, options.get(option))
+
+        if self.zenodo:
+            from boutiques.puller import Puller
+            puller = Puller(desc, self.debug, False)
+            self.desc_dict = json.loads(puller.pull().read())
+        else:
+            # Parse JSON descriptor
+            self.desc_dict = loadJson(desc)
```

Comment:
The logic above could be factorized in a function called by whoever needs to access a descriptor, as mentioned earlier.